site/src/hooks/hookPolyfills.ts

```diff
@@ -35,7 +35,10 @@ export function useEffectEvent<TArgs extends unknown[], TReturn = unknown>(
   callback: (...args: TArgs) => TReturn,
 ) {
   const callbackRef = useRef(callback);
-  useEffect(() => {
+
+  // useLayoutEffect should be overkill here 99% of the time, but it ensures it
+  // will run before any other layout effects that need this custom hook
```

Comment:
eh, you're just describing what `useLayoutEffect` is for. maybe describe why it matters that this runs as a layout effect instead.